### PR TITLE
Add `listClassesInFile` and `listClassesInFiles` functions.

### PR DESCRIPTION
**Pull request by azjezz**

### CODE REVIEW

**Comment by azjezz** [comment]
@WyriHaximus :wave:

**Comment by azjezz** [comment]
>fix appveyor build
>travic ci build broke 
>fix travic ci build
>appveyor build broke 

ugh, i don't why this keeps happening, but i'm getting the same result as appveyor locally

**Comment by WyriHaximus** [comment]
@azjezz 👋

**Comment by azjezz** [comment]
can please check why appveyor and Travis are not getting the same result ? the result seems to be different, locally i get the same result as appveyor, i don't how can i make it pass on both

**Comment by WyriHaximus** [comment]
@azjezz yeah sorry was looking into that, got distracted by something else and then started playing the settlers :zipper_mouth_face: .

Anyway the reason this happens is rather simple. Filesystems don't really care or even store order of nodes in a directory. Your OS (G) takes care of that, so the easiest way is to sort the classes you get from the iterator before asserting you get everything you expect.

**Comment by azjezz** [comment]
@WyriHaximus sorry for the late response, i get distracted last night. 
so what do you think about this ?

File: src/functions.php [hunk 1]
```diff
@@ -44,3 +45,42 @@ function listClassesInDirectory(string $directory): iterable
...
+ * get a list of all classes in the given files.
+ *
+ * @param string[] $files
+ *
+ * @return iterable
+ */
+function listClassesInFiles(string ...$files): iterable
+{
+    foreach ($files as $file) {
+        foreach (listClassesInFile($file) as $class) {
```
Comment (WyriHaximus) [inline src/functions.php:82]: You could use `yield from` here instead: 

```suggestion
        yield from listClassesInFile($file);
```

File: src/functions_include.php [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 namespace WyriHaximus;
 
+use function function_exists;
```
Comment (WyriHaximus) [inline src/functions_include.php:5]: 👍

File: test-app/Foo/Bar/BarAndFoo.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Test\App\Foo\Bar;
```
Comment (WyriHaximus) [inline test-app/Foo/Bar/BarAndFoo.php:3]: Why not put this in the test namespace?

**Review by WyriHaximus — CHANGES_REQUESTED** [review]

File: src/functions.php (hunk 1 above)
Comment (azjezz) [reply src/functions.php:82]: :+1:

File: test-app/Foo/Bar/BarAndFoo.php (hunk 3 above)
Comment (azjezz) [reply test-app/Foo/Bar/BarAndFoo.php:3]: i don't really see the difference here

**Comment by azjezz** [comment]
@WyriHaximus build fail while using `yield from`, can you check that out ?

**Comment by WyriHaximus** [comment]
> @WyriHaximus build fail while using `yield from`, can you check that out ?

Sure :+1:

File: test-app/Foo/Bar/BarAndFoo.php (hunk 3 above)
Comment (WyriHaximus) [reply test-app/Foo/Bar/BarAndFoo.php:3]: Nvm my bad, missed out we put a test app in here when switching context from another project :zipper_mouth_face:

**Comment by WyriHaximus** [comment]
> > @WyriHaximus build fail while using `yield from`, can you check that out ?
> 
> Sure 👍

Tried to figure out, just reverted it to what you initially had as that works and I have no clue how to solve this right now.

**Comment by azjezz** [comment]
@WyriHaximus everything is good ?

**Review by WyriHaximus — APPROVED** [review]

**Comment by WyriHaximus** [comment]
Yes everything looks good 👍 . I'll open a `Hacktoberfest` issue for that `yield from` and see if someone wants to look it.

**Comment by azjezz** [comment]
@WyriHaximus  :clap: :raised_hands:

**Pull request merged**